File: MATH/PrimeNumbers/prime.java

/*
Given a number N, find all prime numbers upto N ( N included ).

Example:

if N = 7,

all primes till 7 = {2, 3, 5, 7}

Make sure the returned array is sorted.
*/


public class Solution {
	public ArrayList<Integer> sieve(int a) {
	    
	    ArrayList<Integer> res=new ArrayList<Integer>();
        int num=0,j=0;
        for(int i=1;i<=a;i++){
            int count=0;
            for(num=i;num>=1;num--){
                if(i%num==0)
                    count+=1;
            }
            if(count==2){
                res.add(i);
                //j+=1;
            }
	}
	return res;
}
}

File: MATH/PrimeNumbers/prime.py

/*
Given a number N, find all prime numbers upto N ( N included ).

Example:

if N = 7,

all primes till 7 = {2, 3, 5, 7}

Make sure the returned array is sorted.
*/


class Solution:
    # @param A : integer
    # @return a list of integers
    def sieve(self, A):
        num=0
        res=[]
        for i in range(1,A+1):
            count=0
            for num in range(i,0,-1):
                if i%num==0:
                    count+=1
            if count==2:
                res.append(i)
                
        return res
                
            
